target/linux/ath79/dts/ar7242_meraki_mr12.dts

```diff
@@ -0,0 +1,180 @@
+#include "ar7242.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "meraki,mr12", "qca,ar7242";
+	model = "Meraki MR12";
+
+	aliases {
+		led-boot = &led_power_orange;
+		led-failsafe = &led_power_orange;
+		led-running = &led_power_green;
+		led-upgrade = &led_power_orange;
+	};
+
+	extosc: ref {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-output-names = "ref";
+		clock-frequency = <40000000>;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		wifi1 {
+			label = "green:wifi1";
+			gpios = <&gpio 11 GPIO_ACTIVE_LOW>;
+		};
+
+		wifi2 {
+			label = "green:wifi2";
+			gpios = <&gpio 12 GPIO_ACTIVE_LOW>;
+		};
+
+		wifi3 {
+			label = "green:wifi3";
+			gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+		};
+
+		wifi4 {
+			label = "green:wifi4";
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+		};
+
+		wan {
+			label = "green:wan";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+		};
+
+		led_power_orange: power_orange {
+			label = "orange:power";
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+			panic-indicator;
+		};
+
+		led_power_green: power_green {
+			label = "green:power";
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 8 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&pcie {
+	status = "okay";
+
+	ath9k: wifi@0,0 {
+		reg = <0x0000 0 0 0 0>;
+		qca,no-eeprom;
+		mtd-mac-address = <&config 0x66>;
+		mtd-mac-address-increment = <1>;
+		#gpio-cells = <2>;
+		gpio-controller;
+	};
+};
+
+&pll {
+	clocks = <&extosc>;
+};
+
+&uart {
+	status = "okay";
+};
+
+&mdio0 {
+	status = "okay";
+
+	phy4: ethernet-phy@4 {
+		reg = <4>;
+	};
+};
+
+&eth0 {
+	status = "okay";
+	mtd-mac-address = <&config 0x66>;
+
+        pll-data = <0x16000000 0x00000101 0x00001313>;
+        pll-reg = <0x4 0x2c 17>;
+        pll-handle = <&pll>;
+
+	phy-mode = "rgmii-id";
+	phy-handle = <&phy4>;
+};
+
+&mdio1 {
+	status = "okay";
+
+	phy0: ethernet-phy@0 {
+		reg = <0>;
+	};
+};
+
+&eth1 {
+	status = "okay";
+	mtd-mac-address = <&config 0x66>;
+
+	phy-handle = <&phy0>;
+        fixed-link {
+		speed = <100>;
+		full-duplex;
+	};
```

Comment:
Strangely enough, without this, the second ethernet port comes up and shows 1000Mbit/s even if nothing is connected to it, at least in dmesg / swconfig. 

Looking at page 24 of the AR7242 datasheet I linked to in the original post, I think I can see why -- the AR7242 has two GMACs, with GMAC1 connecting internally via GMII to the MAC block of the FE switch, all in silicon apparently.

I need to test if this actually causes problems in the interface or on the switch; even still, I don't want users to see a 1000Mbit/s on the FE port, which is why I wanted to set it as fixed 100.

Reply:
Yeah this is normal, you need to imagine the internal switch as a separate device. the link will always be up and gigabit class because it is a link between internal hardware to internal hardware over GMII. For boards that have `ucidef_add_switch`, you can edit VLAN for each port on the switch, and the port connected to the SOC through GMAC1 is called the "CPU" port in LuCI.

As long as you can ping its fine, you might have to add switch configuration otherwise in the 02_network script.

Use swconfig to see which port is the CPU port and which port is the real physical jack: when ethernet is disconnected there will be 1 link up, when it's connected there will be 2
`swconfig dev switch0 show`

Reply:
Thanks for all the help! I think everything works now; I've pushed the new version.